### DISCUSSION TITLE
Unable to successfully read anything on new v4.0 board

### DISCUSSION THREAD

**Discussion by PXAbstraction** [keirf/greaseweazle#387] [Q&A]
Hey all. I'm at my wit's end here and I'm hoping someone might have an answer for me. The fine folks at the VCF forums have been trying to help me out and[ there's a lot of information in the thread over there](https://forum.vcfed.org/index.php?threads/guidance-on-ripping-atari-st-disks-with-greaseweazel.1245642/). But, I'll try to give a synopsis here.

I have a v4.0 Greaseweazel board I purchased on eBay. I've got a new old stock 3.5" drive and a Teac FS-55GFR 5.25" drive. The GW has the most current firmware. I'm on Windows 11 Pro. I'm using the FluxMyFluffyFloppy GUI for the GW tools, but have tried the straight command-line as well.

Initially, I tried ripping Atari ST disks (the main reason I bought it), but have also tried ripping Commodore 64 5.25" disks and regular DOS 3.5" disks. Nothing has worked. No matter what I try to rip, I get variants of this:

> Unknown mark 42
> T78.0: Ignoring unexpected sector C:79 H:0 R:10 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:17 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:1 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:18 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:13 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:14 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:15 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:16 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:11 N:2
> T78.0: Ignoring unexpected sector C:79 H:0 R:12 N:2
> T78.0: IBM MFM (0/18 sectors) from Raw Flux (1624582 flux in 3996.94ms) (Retry
> T78.0: Giving up: 18 sectors missing

So far, I've tried the following:

- New USB cable
- Brand new floppy cable
- Running the drives in sequence and each independently
- Atari ST, DOS and Commodore 64 formatted disks
- Multiple USB ports on my main machine, but I've also tried my laptop

I'm hoping I don't have defective hardware here, but I am honestly at a loss of what to do now. I know this isn't just bad disks because I get the same constant errors, regardless of the disks or the drive.

Does anyone have any steps I could pursue to try to figure this out? Thank you in advance.

**Comment by keirf** [comment]
Firstly if you can, ditch the GUIs for now and test gw on the command line. You're bypassing a layer of abstraction this way and gw command line isn't too unfriendly.

Maybe concentrate on one type of disk for now, say PC 1.44M 3.5" disks. And concentrate on reading a couple of cylinders during testing, rather than a whole disk:
```gw read --tracks=c=0-1 --format=ibm.1440 name.img```

You may need `--drive=b` if your cable doesn't have a twist. `name.img` is just a name for the generated IMG file.

Your errors look like you are off by one cylinder in your reads. Note unexpected sector reports with cylinder (C) reported as 79 when Greaseweazle is trying to read cylinder 78 (for example). That's the first thing to sort out. I would say that's a drive calibration error unless you see the same on multiple 3.5 inch drives?

**Comment by PXAbstraction** [comment]
Hi there and thanks for replying! Sorry for my delays, holidays and whatnot. :)

The GUIs just generate and pass command lines anyway, but I get your point about removing abstraction so I did try that, but sadly got the same results, at least when using the same commands.

I updated my GW to the latest firmware and then ran the command you got using two different 1.44MB PC floppy disks. These are the results I got:

C:\COMSTUFF\Greaseweazle>gw read --tracks=c=0-1 --format=ibm.1440 "c:\comstuff\no save\test.img"
Reading c=0-1:h=0-1 revs=2
Format ibm.1440
T0.0: IBM MFM (0/18 sectors) from Raw Flux (157455 flux in 400.08ms)
T0.0: IBM MFM (0/18 sectors) from Raw Flux (393678 flux in 999.96ms) (Retry #1.1)
T0.0: IBM MFM (0/18 sectors) from Raw Flux (629841 flux in 1599.79ms) (Retry #1.2)
T0.0: IBM MFM (0/18 sectors) from Raw Flux (866003 flux in 2199.62ms) (Retry #1.3)
T0.0: Giving up: 18 sectors missing
T0.1: IBM MFM (2/18 sectors) from Raw Flux (183322 flux in 400.05ms)
T0.1: IBM MFM (2/18 sectors) from Raw Flux (458170 flux in 999.88ms) (Retry #1.1)
T0.1: IBM MFM (2/18 sectors) from Raw Flux (733027 flux in 1599.71ms) (Retry #1.2)
T0.1: IBM MFM (2/18 sectors) from Raw Flux (1007874 flux in 2199.52ms) (Retry #1.3)
T0.1: Giving up: 16 sectors missing
T1.0: IBM MFM (1/18 sectors) from Raw Flux (156256 flux in 400.04ms)
T1.0: IBM MFM (1/18 sectors) from Raw Flux (390534 flux in 999.84ms) (Retry #1.1)
T1.0: IBM MFM (1/18 sectors) from Raw Flux (624786 flux in 1599.65ms) (Retry #1.2)
T1.0: IBM MFM (1/18 sectors) from Raw Flux (859071 flux in 2199.46ms) (Retry #1.3)
T1.0: Giving up: 17 sectors missing
T1.1: IBM MFM (2/18 sectors) from Raw Flux (158831 flux in 400.03ms)
T1.1: IBM MFM (2/18 sectors) from Raw Flux (396912 flux in 999.84ms) (Retry #1.1)
T1.1: IBM MFM (2/18 sectors) from Raw Flux (635027 flux in 1599.65ms) (Retry #1.2)
T1.1: IBM MFM (2/18 sectors) from Raw Flux (873079 flux in 2199.44ms) (Retry #1.3)
T1.1: Giving up: 16 sectors missing
Cyl-> 0
H. S: 01
0. 0: XX
0. 1: XX
0. 2: XX
0. 3: X.
0. 4: XX
0. 5: XX
0. 6: XX
0. 7: XX
0. 8: XX
0. 9: XX
0.10: XX
0.11: XX
0.12: XX
0.13: XX
0.14: XX
0.15: XX
0.16: XX
0.17: XX
1. 0: XX
1. 1: XX
1. 2: XX
1. 3: ..
1. 4: ..
1. 5: XX
1. 6: XX
1. 7: XX
1. 8: XX
1. 9: XX
1.10: XX
1.11: XX
1.12: XX
1.13: XX
1.14: XX
1.15: XX
1.16: XX
1.17: XX
Found 5 sectors of 72 (6%)

C:\COMSTUFF\Greaseweazle>gw read --tracks=c=0-1 --format=ibm.1440 "c:\comstuff\no save\test1.img"
Reading c=0-1:h=0-1 revs=2
Format ibm.1440
T0.0: IBM MFM (0/18 sectors) from Raw Flux (161651 flux in 399.95ms)
T0.0: IBM MFM (0/18 sectors) from Raw Flux (403623 flux in 999.74ms) (Retry #1.1)
T0.0: IBM MFM (0/18 sectors) from Raw Flux (645515 flux in 1599.51ms) (Retry #1.2)
T0.0: IBM MFM (0/18 sectors) from Raw Flux (887195 flux in 2199.30ms) (Retry #1.3)
T0.0: Giving up: 18 sectors missing
Unknown mark ff
T0.1: IBM MFM (0/18 sectors) from Raw Flux (182837 flux in 400.05ms)
Unknown mark ff
Unknown mark ff
Unknown mark ff
T0.1: IBM MFM (0/18 sectors) from Raw Flux (457021 flux in 999.83ms) (Retry #1.1)
Unknown mark ff
Unknown mark ff
Unknown mark ff
T0.1: IBM MFM (0/18 sectors) from Raw Flux (731319 flux in 1599.61ms) (Retry #1.2)
Unknown mark ff
Unknown mark ff
T0.1: IBM MFM (0/18 sectors) from Raw Flux (1005474 flux in 2199.42ms) (Retry #1.3)
T0.1: Giving up: 18 sectors missing
T1.0: IBM MFM (0/18 sectors) from Raw Flux (158421 flux in 400.01ms)
T1.0: IBM MFM (0/18 sectors) from Raw Flux (396066 flux in 999.79ms) (Retry #1.1)
T1.0: IBM MFM (0/18 sectors) from Raw Flux (633496 flux in 1599.59ms) (Retry #1.2)
T1.0: IBM MFM (0/18 sectors) from Raw Flux (870853 flux in 2199.38ms) (Retry #1.3)
T1.0: Giving up: 18 sectors missing
T1.1: IBM MFM (0/18 sectors) from Raw Flux (155479 flux in 400.02ms)
T1.1: IBM MFM (0/18 sectors) from Raw Flux (388590 flux in 999.81ms) (Retry #1.1)
T1.1: IBM MFM (0/18 sectors) from Raw Flux (621717 flux in 1599.59ms) (Retry #1.2)
T1.1: IBM MFM (0/18 sectors) from Raw Flux (854825 flux in 2199.37ms) (Retry #1.3)
T1.1: Giving up: 18 sectors missing
Cyl-> 0
H. S: 01
0. 0: XX
0. 1: XX
0. 2: XX
0. 3: XX
0. 4: XX
0. 5: XX
0. 6: XX
0. 7: XX
0. 8: XX
0. 9: XX
0.10: XX
0.11: XX
0.12: XX
0.13: XX
0.14: XX
0.15: XX
0.16: XX
0.17: XX
1. 0: XX
1. 1: XX
1. 2: XX
1. 3: XX
1. 4: XX
1. 5: XX
1. 6: XX
1. 7: XX
1. 8: XX
1. 9: XX
1.10: XX
1.11: XX
1.12: XX
1.13: XX
1.14: XX
1.15: XX
1.16: XX
1.17: XX
Found 0 sectors of 72 (0%)

These are retail disks from a boxed copy of X-Wing that I have. The first disk got 6% of sectors apparently and the second for 0%. 

For kicks, I reformatted the command and tried a 1.2MB disk from my Teac drive and got this:

C:\COMSTUFF\Greaseweazle>gw read --tracks=c=0-1 --drive=b --format=ibm.1200 "c:\comstuff\no save\test1.img"
Reading c=0-1:h=0-1 revs=2
Format ibm.1200
T0.0: IBM MFM (0/15 sectors) from Raw Flux (63249 flux in 334.33ms)
T0.0: IBM MFM (0/15 sectors) from Raw Flux (158067 flux in 835.57ms) (Retry #1.1)
T0.0: IBM MFM (0/15 sectors) from Raw Flux (252885 flux in 1336.80ms) (Retry #1.2)
T0.0: IBM MFM (0/15 sectors) from Raw Flux (347701 flux in 1838.02ms) (Retry #1.3)
T0.0: Giving up: 15 sectors missing
T0.1: IBM MFM (0/15 sectors) from Raw Flux (128164 flux in 334.31ms)
T0.1: IBM MFM (0/15 sectors) from Raw Flux (320099 flux in 835.53ms) (Retry #1.1)
T0.1: IBM MFM (0/15 sectors) from Raw Flux (511977 flux in 1336.74ms) (Retry #1.2)
T0.1: IBM MFM (0/15 sectors) from Raw Flux (703884 flux in 1837.95ms) (Retry #1.3)
T0.1: Giving up: 15 sectors missing
T1.0: IBM MFM (0/15 sectors) from Raw Flux (68400 flux in 334.30ms)
T1.0: IBM MFM (0/15 sectors) from Raw Flux (171020 flux in 835.51ms) (Retry #1.1)
T1.0: IBM MFM (0/15 sectors) from Raw Flux (273524 flux in 1336.71ms) (Retry #1.2)
T1.0: IBM MFM (0/15 sectors) from Raw Flux (376165 flux in 1837.91ms) (Retry #1.3)
T1.0: Giving up: 15 sectors missing
T1.1: IBM MFM (0/15 sectors) from Raw Flux (118669 flux in 334.30ms)
T1.1: IBM MFM (0/15 sectors) from Raw Flux (296338 flux in 835.50ms) (Retry #1.1)
T1.1: IBM MFM (0/15 sectors) from Raw Flux (473796 flux in 1336.70ms) (Retry #1.2)
T1.1: IBM MFM (0/15 sectors) from Raw Flux (651355 flux in 1837.90ms) (Retry #1.3)
T1.1: Giving up: 15 sectors missing
Cyl-> 0
H. S: 01
0. 0: XX
0. 1: XX
0. 2: XX
0. 3: XX
0. 4: XX
0. 5: XX
0. 6: XX
0. 7: XX
0. 8: XX
0. 9: XX
0.10: XX
0.11: XX
0.12: XX
0.13: XX
0.14: XX
1. 0: XX
1. 1: XX
1. 2: XX
1. 3: XX
1. 4: XX
1. 5: XX
1. 6: XX
1. 7: XX
1. 8: XX
1. 9: XX
1.10: XX
1.11: XX
1.12: XX
1.13: XX
1.14: XX
Found 0 sectors of 60 (0%)

Basically the same results. This 3.5" was brand new in box, but it's also a more "modern" drive (it advertised supporting Windows Vista on the box). I don't have another one, but I can see if I can track one down. I'm surprised that I'm getting the exact same result on a totally different drive though, that was sold to me as tested. Do you think that's still the root cause here or could it possibly be something else?

Thanks again. :)

**Reply by keirf** [reply]
The latter you say is a 1.2mb disk but you mention only 3.5 inch drives. Is a 5.25 inch drive in play too? 3.5 inch high density disks are usually 1.44mb.

Take a full raw dump of a known good disk on a known good drive (if you have these) and upload to a cloud drive and link here for download so I can take a look. To do this just dump to an scp image file and do not specify a `--format`

**Reply by PXAbstraction** [reply]
Yeah, I have a Teac FD-55GFR as well. The drives are daisy chained on the same twisted ribbon cable, but I have tried only having one connected at a time and gotten the same results. The Teac is as known good as I can get as it was tested by the seller I purchased it from. I have tried a number of disks in both drives so I'd like to think at least one of them is known good, but I'll double-check and see what I can find and post the images as you requested. Thanks! :)

**Reply by PXAbstraction** [reply]
Hey there.

Well, I have to say I'm stunned. That brand new 3.5" drive I bought apparently is either defective or just doesn't work with my GW for some reason. A friend had an old Panasonic drive lying around that he lent me and at least with PC 1.44MB disks, it's working perfectly. My Teac 5.25" drive still isn't though, which is concerning, but I'm going to find some more disks to test with it. Progress! I'll report back as I learn more.

Thanks!

**Reply by PXAbstraction** [reply]
Welp, the problem definitely appears to be related to the Teac drive somehow. I've had some people on the VCF forums helping me verify that it's properly jumpered and whatnot, but it still fails to read anything, even though it does communicate with the GW and you can see it spinning and hear the head moving. This drive was supposed to be tested as working before I bought it so I'm not sure if it's a defect or something else. However, it's definitely not to do with the GW so I'll keep pursuing that angle instead.

Thank you for your help and your work on this awesome tool!

**Reply by keirf** [reply]
For the Teac, can you please read to a raw flux file and attach for inspection? Something like
```
gw read --tracks=c=0-1 --drive=b name.scp
```

Another thing to try is your original command line, but try adding `--dd=L` to the command line. The drive may need to be signalled about the disk density, and this extra command line option will do that.

EDIT: You could also try dumping a raw flux file both with and without `--dd=L`, and upload both for inspection.

**Reply by PXAbstraction** [reply]
[RAW Flux Tests.zip](https://github.com/keirf/greaseweazle/files/13876830/RAW.Flux.Tests.zip)

Here's the tests you asked for. Note that this is from a known good Commodore disk as I haven't been able to find a 1.2MB DOS disk that I'm confident is in good shape. Using the --dd parameter unfortunately didn't help either.

I have a friend who has an old Commodore PC machine with a 5.25" in it he's willing to take out and lend me. I have no idea what brand or model it is, but hopefully it can give a basis for comparison.

**Reply by keirf** [reply]
The good news is that track 0.0 of those two SCP image files are both valid C64 tracks! So I think your drive works okay.

Here is the correct command line for reading a C64 disk:
```
gw read --drive=b --tracks=step=2 name.d64
```

Using `.d64` will automatically select the `commodore.1541` disk format. You need to specify `step=2` to double-step the drive heads because you are reading a 40-track disk in an 80-track drive.

In short this means that your drive is probably fine. You just need a good 1.2MB disk to test! And you should consider trying such prospective disks with and without `--dd=L`

**Reply by keirf** [reply]
Here's another trick. You can dump any disk, all tracks like this:
```
gw read --drive=b name.scp
```

And then you can try to decode it as a separate step, for example like this:
```
gw convert --tracks=step=2 name.scp name.d64
```

Note that `gw convert` is a lot like an offline version of `gw read`. It imports an existing image file and does not require Greaseweazle hardware to be attached.

**Reply by PXAbstraction** [reply]
So, I tried to read a few C64 disks using this command:

"C:\COMSTUFF\Greaseweazle\gw.exe" read --format commodore.1541 --tracks=c=0-79:h=0:step=2 --revs=5 "C:\COMSTUFF\No Save\test.d64"

A couple were retail disks, a couple were just regular disks. This is the result I got from them all:

"C:\COMSTUFF\Greaseweazle\gw.exe" 
79:h=0:step=2 --revs=5 "C:\COMSTUFF\No Save\test.d64"
Reading c=0-79:h=0:step=2 revs=5
Format commodore.1541
T0.0: Commodore GCR (21/21 sectors) from Raw Flux (191176 flux in 903.81ms)
T1.0: Commodore GCR (21/21 sectors) from Raw Flux (175532 flux in 911.16ms)
T2.0: Commodore GCR (21/21 sectors) from Raw Flux (188660 flux in 917.01ms)
T3.0: Commodore GCR (21/21 sectors) from Raw Flux (185781 flux in 915.86ms)
T4.0: Commodore GCR (21/21 sectors) from Raw Flux (208322 flux in 914.70ms)
T5.0: Commodore GCR (21/21 sectors) from Raw Flux (176186 flux in 914.42ms)
T6.0: Commodore GCR (21/21 sectors) from Raw Flux (174997 flux in 917.83ms)
T7.0: Commodore GCR (21/21 sectors) from Raw Flux (173584 flux in 918.26ms)
T8.0: Commodore GCR (21/21 sectors) from Raw Flux (190709 flux in 926.04ms)
T9.0: Commodore GCR (21/21 sectors) from Raw Flux (176746 flux in 924.62ms)
T10.0: Commodore GCR (21/21 sectors) from Raw Flux (178277 flux in 925.83ms)
T11.0: Commodore GCR (21/21 sectors) from Raw Flux (190587 flux in 924.65ms)
T12.0: Commodore GCR (21/21 sectors) from Raw Flux (206936 flux in 925.09ms)
T13.0: Commodore GCR (21/21 sectors) from Raw Flux (205614 flux in 922.31ms)
T14.0: Commodore GCR (21/21 sectors) from Raw Flux (196818 flux in 923.52ms)
T15.0: Commodore GCR (21/21 sectors) from Raw Flux (198116 flux in 923.82ms)
T16.0: Commodore GCR (21/21 sectors) from Raw Flux (198058 flux in 919.67ms)
T17.0: Commodore GCR (19/19 sectors) from Raw Flux (178314 flux in 924.24ms)
T18.0: Commodore GCR (19/19 sectors) from Raw Flux (165532 flux in 927.51ms)
T19.0: Commodore GCR (19/19 sectors) from Raw Flux (167093 flux in 928.22ms)
T20.0: Commodore GCR (19/19 sectors) from Raw Flux (165662 flux in 929.40ms)
T21.0: Commodore GCR (19/19 sectors) from Raw Flux (180639 flux in 929.00ms)
T22.0: Commodore GCR (19/19 sectors) from Raw Flux (184350 flux in 925.36ms)
T23.0: Commodore GCR (19/19 sectors) from Raw Flux (190909 flux in 925.90ms)
T24.0: Commodore GCR (18/18 sectors) from Raw Flux (170053 flux in 925.26ms)
T25.0: Commodore GCR (18/18 sectors) from Raw Flux (164323 flux in 926.11ms)
T26.0: Commodore GCR (18/18 sectors) from Raw Flux (171957 flux in 929.19ms)
T27.0: Commodore GCR (18/18 sectors) from Raw Flux (153345 flux in 928.35ms)
T28.0: Commodore GCR (18/18 sectors) from Raw Flux (160980 flux in 930.91ms)
T29.0: Commodore GCR (18/18 sectors) from Raw Flux (175328 flux in 930.42ms)
T30.0: Commodore GCR (17/17 sectors) from Raw Flux (156773 flux in 928.73ms)
T31.0: Commodore GCR (17/17 sectors) from Raw Flux (151338 flux in 931.72ms)
T32.0: Commodore GCR (17/17 sectors) from Raw Flux (166982 flux in 931.74ms)
T33.0: Commodore GCR (17/17 sectors) from Raw Flux (167986 flux in 931.43ms)
T34.0: Commodore GCR (17/17 sectors) from Raw Flux (151409 flux in 931.19ms)
T35.0: Commodore GCR (0/17 sectors) from Raw Flux (211970 flux in 932.79ms)
T35.0: Commodore GCR (0/17 sectors) from Raw Flux (421967 flux in 1856.98ms) 
T35.0: Commodore GCR (0/17 sectors) from Raw Flux (631590 flux in 2779.51ms) 
T35.0: Commodore GCR (0/17 sectors) from Raw Flux (841655 flux in 3704.00ms) 
T35.0: Giving up: 17 sectors missing
T36.0: Commodore GCR (0/17 sectors) from Raw Flux (201172 flux in 885.31ms)
T36.0: Commodore GCR (0/17 sectors) from Raw Flux (411504 flux in 1811.08ms) 
T36.0: Commodore GCR (0/17 sectors) from Raw Flux (621899 flux in 2737.12ms) 
T36.0: Commodore GCR (0/17 sectors) from Raw Flux (831886 flux in 3661.35ms) 
T36.0: Giving up: 17 sectors missing
T37.0: Commodore GCR (0/17 sectors) from Raw Flux (200751 flux in 883.51ms)
T37.0: Commodore GCR (0/17 sectors) from Raw Flux (412237 flux in 1814.30ms) 
T37.0: Commodore GCR (0/17 sectors) from Raw Flux (619475 flux in 2726.35ms) 
T37.0: Commodore GCR (0/17 sectors) from Raw Flux (829324 flux in 3650.03ms) 
T37.0: Giving up: 17 sectors missing
T38.0: Commodore GCR (0/17 sectors) from Raw Flux (197241 flux in 867.90ms)
T38.0: Commodore GCR (0/17 sectors) from Raw Flux (408415 flux in 1797.06ms) 
T38.0: Commodore GCR (0/17 sectors) from Raw Flux (614668 flux in 2704.68ms) 
T38.0: Commodore GCR (0/17 sectors) from Raw Flux (824628 flux in 3628.64ms) 
T38.0: Giving up: 17 sectors missing
T39.0: Commodore GCR (0/17 sectors) from Raw Flux (200653 flux in 882.95ms)
T39.0: Commodore GCR (0/17 sectors) from Raw Flux (410938 flux in 1808.45ms) 
T39.0: Commodore GCR (0/17 sectors) from Raw Flux (620870 flux in 2732.40ms) 
T39.0: Commodore GCR (0/17 sectors) from Raw Flux (830292 flux in 3654.06ms) 
T39.0: Giving up: 17 sectors missing
T40.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T41.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T42.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T43.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T44.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T45.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T46.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T47.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T48.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T49.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
T50.0: WARNING: Out of range for format 'commodore.1541': No format conversion 
Command Failed: Seek: Invalid cylinder 102

Once the warnings starting at track 40 began, the drive would make what sounded like a head bang before each of those lines was displayed. Sounds like it ran out of space to travel. It also ended up with no .d64 file at the end.

The same friend who has the other floppy drive also has some 1.2MB disks, so I'm going to try those either tonight or tomorrow, depending on when the blizzard outside ends. :)

**Reply by keirf** [reply]
The main issue here is that you should not specify `c=0-79:h=0`. The correct cylinder and head range will be selected for D64 image, which is 35 cylinders, single sided (ie. `c=0-34:h=0`). What *you* have specified is 80 cylinders and double stepping, hence Greaseweazle will attempt to get to cylinder 160, which of course doesn't exist! Many drives will silently reject seeks beyond their limit, but clearly not yours. What yours is doing is banging the drive head against the drive chassis. As you can imagine, this is not good for your drive's calibration!

You also don't need to specify `--revs=` in concert with a sector image format. Because Greaseweazle is doing track decode, it knows when it has a good or bad track, and will insert retries automatically. So `--revs=` is mainly useful for dumb raw reads, for example to raw flux files (such as SCP) with no `--format` specified.

Oh and you don't need `--format=` either, as it is implied by creating a D64 image file.

So your correct command line would be:
```
"C:\COMSTUFF\Greaseweazle\gw.exe" read --tracks=step=2 "C:\COMSTUFF\No Save\test.d64"
```

**Reply by PXAbstraction** [reply]
Hey again! With all your amazing advice and the help of the VCF forums, I think I have everything figured out. With the new (old) drive, I'm now able to successfully rip 3.5" disks and now that I know about the extra commands for double-stepping and the like, I think I'll be able to rip the 5.25" disks I want with the Teac drive as well.

I will say, the GW is incredibly powerful, but it definitely can be tricky for someone like me to start out who is knowledgeable, but not an expert in the nuances of how disk drives work. :) I run a small YouTube channel where I sometimes do retro content and I'm considering putting a couple of videos together about getting started with a GW using some of the more common retro formats. Maybe that would help give more lamen users a good place to begin.

Regardless, I thank you for your help and patience and I'm really excited to use this to get started on the archival projects I have planned. Cheers and thanks for making something so useful!

**Reply by wohali** [reply]
For a simpler process (esp. for beginners), you can also consider using something like the [FluxEngine UI](https://cowlark.com/fluxengine/index.html), which supports GW hardware.

**Reply by PXAbstraction** [reply]
I actually have tried that, but ran into many of the same problems I had here. Just needed more core knowledge in the end.

**Reply by keirf** [reply]
Yes Greaseweazle is painfully lacking in tutorial material. I recently had a new Getting Started wiki page contributed, which is a lot better than I had before. But more material would be amazing.
